op-node/rollup/derive/channel_test.go

```diff
@@ -99,3 +106,60 @@ func TestFrameValidity(t *testing.T) {
 		t.Run(tc.name, tc.Run)
 	}
 }
+
+func TestBatchReader(t *testing.T) {
+	// Get batch data
+	rng := rand.New(rand.NewSource(0x543331))
+	singularBatch := RandomSingularBatch(rng, 5000, big.NewInt(333))
+	batchDataInput := NewBatchData(singularBatch)
+
+	encodedBatch := &bytes.Buffer{}
+	// Get the encoded data of the batch data
+	batchDataInput.encodeTyped(encodedBatch)
+
+	var testCases = []struct {
+		name string
+		algo func(buf *bytes.Buffer)
+	}{{
+		name: "zlib",
+		algo: func(buf *bytes.Buffer) {
+			writer := zlib.NewWriter(buf)
+			writer.Write(encodedBatch.Bytes())
+		},
+	},
+		{
+			name: "brotli10",
+			algo: func(buf *bytes.Buffer) {
+				buf.WriteByte(ChannelVersionBrotli)
+				writer := brotli.NewWriterLevel(buf, 10)
+				writer.Write(encodedBatch.Bytes())
+			},
+		}, {
+			name: "zstd",
+			algo: func(buf *bytes.Buffer) {
+				writer := zstd.NewWriter(buf)
+				writer.Write(encodedBatch.Bytes())
+			},
+		}}
+
+	for _, tc := range testCases {
+		compressed := bytes.NewBuffer([]byte{})
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			tc.algo(compressed)
+			reader, err := BatchReader(bytes.NewReader(compressed.Bytes()))
+			if tc.name == "zstd" {
+				require.NotNil(t, err)
+				return
+			}
+			require.Nil(t, err)
```

Comment:
Always use
```suggestion
			require.NoError(t, err)
```
it formats errors better in failed test output, and there are actually edge cases where wrong assertions can be made with `require.Nil`.